### ISSUE TITLE
Issue loading some workspaces

### ISSUE THREAD

**Issue by daddykotex** [awslabs/smithy-language-server#39]
I have not been able to reproduce in the small yet, but we are seeing issues on some of our project when building the latest changes from this repository (061d596cdb8da50acf7bb8423791e3b0838f7402).

In short, when opening a Smithy file, the extension kicks in and start the LSP. The LSP fails to run the `#initialize` method. The result is that we never see the notification about `N jars were loaded and X smithy definitions were found`. It's kind of silent, but the result is that most features don't work properly because `this.project` the the `SmithyTextDocumentService` is null.

This eventually boils down to `public void createProject(SmithyBuildExtensions ext, File root)` that fails with the following error:

With added debugging stuff, I recovered on error:

```
[15:10:34] [15:10:34] java.lang.IndexOutOfBoundsException: Index -2 out of bounds for length 67
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:459)
	at software.amazon.smithy.lsp.ext.SmithyProject.advanceMarkerOnNonMemberShapes(SmithyProject.java:255)
	at software.amazon.smithy.lsp.ext.SmithyProject.collectLocations(SmithyProject.java:233)
	at software.amazon.smithy.lsp.ext.SmithyProject.lambda$new$0(SmithyProject.java:68)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at software.amazon.smithy.lsp.ext.SmithyProject.<init>(SmithyProject.java:68)
	at software.amazon.smithy.lsp.ext.SmithyProject.load(SmithyProject.java:170)
	at software.amazon.smithy.lsp.ext.SmithyProject.load(SmithyProject.java:141)
	at software.amazon.smithy.lsp.SmithyTextDocumentService.createProject(SmithyTextDocumentService.java:119)
	at software.amazon.smithy.lsp.SmithyTextDocumentService.createProject(SmithyTextDocumentService.java:169)
	at software.amazon.smithy.lsp.SmithyLanguageServer.lambda$loadSmithyBuild$0(SmithyLanguageServer.java:63)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at software.amazon.smithy.lsp.SmithyLanguageServer.loadSmithyBuild(SmithyLanguageServer.java:63)
	at software.amazon.smithy.lsp.SmithyLanguageServer.initialize(SmithyLanguageServer.java:72)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

I'll update this Issue with a smaller repro if I can find one. But I decided to go ahead and open as it is as you could know how to fix that 😛 

-----

**Update**

It seems like the problem happens when `advanceMarkerOnNonMemberShapes` is called to find the end position. If the location returned by [`traits.get(0).getSourceLocation().getLine()`](https://github.com/awslabs/smithy-language-server/pull/35/files#diff-df12d2929de074756638003082c75f66144f9372a43972cb26280f3b9bb0115aR231) is `SourceLocation.NONE`, then all hell breaks loose!

**Comment by daddykotex** [comment]
I gave it a shot and it seems like the problem is gone and the project loads correctly.

I may have found another issue. I'll open a separate issue.

**Comment by daddykotex** [comment]
Fixed by #40
